awpy/visualization/plot.py

```diff
@@ -326,3 +346,369 @@ def plot_nades(
+    _plot_map_control_snapshot_helper(
+        map_name, tile_values, base_ax, player_data=player_pos
+    )
+    if save_filepath:
+        figure.savefig(fname=save_filepath, bbox_inches="tight", dpi=400)
+        plt.close(figure)
+    elif not given_fig_ax[0]:
+        plt.show()
+        plt.close(figure)
+
+
+def plot_frame_map_control(
+    map_name: str,
+    frame: GameFrame,
+    plot_type: str = "",
+    given_fig_ax: tuple[plt.Figure, plt.Axes] | tuple[None, None] = (None, None),
+    save_filepath: str = "",
+) -> None:
+    """Visualize map control for awpy frame.
+
+    Args:
+        map_name (str): Map used position_transform call
+        frame (GameFrame): awpy frame to calculate map control for
+        plot_type (str): Determines which type of plot is created
+            (either with or without players)
+        given_fig_ax: Optional tuple containing figure and ax objects for plotting
+        save_filepath (str): Filepath to save the figure to file if required
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    player_positions = extract_teams_metadata(frame)
+    map_control_dict = calc_parsed_frame_map_control_values(map_name, player_positions)
+    if plot_type == "players":
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            player_pos=player_positions,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+    else:
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+
+
+def create_round_map_control_gif(
+    map_name: str,
+    round_data: GameRound,
+    plot_type: str = "",
+    gif_filepath: str = "./results/round_mc.gif",
+) -> None:
+    """Create gif summarizing map control for round.
+
+    Args:
+        map_name (str): Map used in plot_frame_map_control call
+        round_data (GameRound): Round whose map control will be animated.
+            Expected format that of awpy round
+        plot_type (str): Determines which type of plot is created
+            (either with or without players)
+        gif_filepath (str): Filepath to save the gif to file
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    ]
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    if os.path.isdir(AWPY_TMP_FOLDER):
+        shutil.rmtree(f"{AWPY_TMP_FOLDER}/")
+    os.mkdir(AWPY_TMP_FOLDER)
+
+    images: list[np.ndarray] = []
+    print("Saving/loading frames")
+    frames = round_data["frames"]
+
+    for i, frame in enumerate(frames or []):
```

Comment:
tqdm could be helpful here since the function is not too fast.

If tqdm is added, add a `verbose` argument for tqdm

Reply:
Addressed in next commit to branch.